lib/cocoapods/validator.rb

```diff
@@ -142,7 +142,12 @@ def failure_reason
         reasons << 'all results apply only to public specs, but you can use ' \
                    '`--private` to ignore them if linting the specification for a private pod'
       end
-      reasons.to_sentence
+      if dot_swift_version.nil?
```

Comment:
can we add some logic similar to the `pod_target.uses_swift` that see's if any of the related targets contain Swift to ensure that is actually a Swift pod?